File: android/src/main/java/io/github/v7lin/fake_proxy/FakeProxyPlugin.java
package io.github.v7lin.fake_proxy;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/**
 * FakeProxyPlugin
 */
public class FakeProxyPlugin implements MethodCallHandler {
    /**
     * Plugin registration.
     */
    public static void registerWith(Registrar registrar) {
        final MethodChannel channel = new MethodChannel(registrar.messenger(), "v7lin.github.io/fake_proxy");
        channel.setMethodCallHandler(new FakeProxyPlugin());
    }

    private static final String METHOD_GETPROXY = "getProxy";

    @Override
    public void onMethodCall(MethodCall call, Result result) {
        if (METHOD_GETPROXY.equals(call.method)) {
            if (Boolean.parseBoolean(System.getProperty("http.proxySet"))) {
                String proxyHost = System.getProperty("http.proxyHost");
                int proxyPort = Integer.parseInt(System.getProperty("http.proxyPort"));
                result.success(String.format("%1$s:%2$d", proxyHost, proxyPort));
            } else {
                result.success(null);
            }
        } else {
            result.notImplemented();
        }
    }
}

File: ios/Classes/FakeProxyPlugin.m
#import "FakeProxyPlugin.h"

@implementation FakeProxyPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    FlutterMethodChannel* channel = [FlutterMethodChannel
                                     methodChannelWithName:@"v7lin.github.io/fake_proxy"
                                     binaryMessenger:[registrar messenger]];
    FakeProxyPlugin* instance = [[FakeProxyPlugin alloc] init];
    [registrar addMethodCallDelegate:instance channel:channel];
}

static NSString * const METHOD_GETPROXY = @"getProxy";

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
    if ([METHOD_GETPROXY isEqualToString:call.method]) {
        CFDictionaryRef proxySettings = CFNetworkCopySystemProxySettings();
        NSDictionary *dictProxy = (__bridge_transfer id)proxySettings;
        //是否开启了http代理
        if ([[dictProxy objectForKey:@"HTTPEnable"] boolValue]) {
            NSString * proxyHost = [dictProxy objectForKey:@"HTTPProxy"]; //代理地址
            int proxyPort = [[dictProxy objectForKey:@"HTTPPort"] intValue];  //代理端口号
            result([NSString stringWithFormat:@"%@:%d", proxyHost, proxyPort]);
        } else {
            result(nil);
        }
    } else {
        result(FlutterMethodNotImplemented);
    }
}

@end
